Add a task to fix locale issues

microstudi opened the pull request:
#### :tophat: What? Why?

In certain cases is necessary to manipulate locales of an organization through the database. This adds a new task namespace and some commands to solve common issues.
It also provides a new documentation file explaining the usage.

#### :clipboard: Subtasks
- [x] Add documentation regarding the feature

mrcasals commented on decidim-core/lib/tasks/decidim_locales_tasks.rake:
```diff
@@ -0,0 +1,24 @@
+    desc "Rebuild the search index"
+    task rebuild_search: :environment do
+      Decidim::SearchableResource.destroy_all
+      Decidim::Searchable.searchable_resources.pluck(0).each do |resource|
```
```suggestion
      Decidim::Searchable.searchable_resources.keys.each do |resource|
```

microstudi replied:
yep!

mrcasals commented on docs/advanced/fixing_locales.md:
```diff
@@ -0,0 +1,84 @@
+
+When you create an organization, you choose the available languages for it (through the `/system/` url). However, when trying to edit it, the language selector is not available anymore. Here is a way to update that locales manually:
+
+First, be sure that your initializer file has all the locales you want:
```
```suggestion
First, make sure that your initializer file has all the locales you want:
```

mrcasals commented on docs/advanced/fixing_locales.md:
````diff
@@ -0,0 +1,84 @@
+bundle exec rake decidim:locales:sync_all
+```
+
+Run this task if you want have changed your `config/initializers/decidim.rb` `available_locales` or `default_locale` and you want to modify all the organizations so they have the same settings.
````
```suggestion
Run this task if you want to have changed `available_locales` or `default_locale` in your `config/initializers/decidim.rb` and you want to modify all the organizations so they have the same settings.
```

microstudi replied:
In here I've rephrase it.

mrcasals reviewed (dismissed):
Nice! I left a few suggestions

mrcasals reviewed (approved):
Looks good to me!

mrcasals commented on decidim-core/lib/tasks/decidim_locales_tasks.rake:
```diff
@@ -0,0 +1,24 @@
+    task rebuild_search: :environment do
+      Decidim::SearchableResource.destroy_all
+      Decidim::Searchable.searchable_resources.keys.each do |resource|
+        resource.constantize.all.each(&:try_update_index_for_search_resource)
```
```suggestion
        resource.constantize.find_each(&:try_update_index_for_search_resource)
```

This is more performant, memory-wise, as it loads the resources in chunks instead of all of them at the same time.

mrcasals replied:
Also, the `Decidim::Searchable.searchable_resources` has the classes as values. I'm not sure if you're constantizing for some specific reason, but the whole block could be changed to this:

```ruby
      Decidim::Searchable.searchable_resources.values.each do |searchable_klass|
        searchable_klass.find_each(&:try_update_index_for_search_resource)
      end
```

microstudi replied:
oh, I don't even remember. I guess I saw the key and my brain executed a `return`!
Thanks for your insights! I'll make the changes!

mrcasals replied:
If you prefer, you can just ignore these lastc hangees and keep only the `find_each` suggestion 😄

microstudi replied:
No, I like it. I'll continue in the afternoon, I have a busy morning.
Thinking out loud, we could add a task to add or remove a language from an organization and perform all the associated task for that organization.

microstudi commented:
@mrcasals in the end, I've refactored a little to show more information when performing the task. Also changed some docs that where wrong. The result of both commands looks like this:
![image](https://user-images.githubusercontent.com/1401520/92760539-094bee00-f391-11ea-9b0d-86b89577ff8f.png)

`Decidim.available_locales` is `:es, :en, :ca`:
![image](https://user-images.githubusercontent.com/1401520/92760977-795a7400-f391-11ea-9d86-4ca45e8e2d05.png)

mrcasals commented:
@microstudi nice!! Good job!

I've restarted a teest run that failede, looked like some Ruby setup problem

Leusev reviewed (approved):
Looks fantastic for me @microstudi 
Liked the idea about what [you pointed out](https://github.com/decidim/decidim/pull/6510#discussion_r486148578) about to add some new tasks in order to add or remove a new language, since it is not a rare thing and it has also been requested on some occasion...
Thanks for your job :smile:
